### DISCUSSION TITLE
Realm not loading from database

### DISCUSSION THREAD

**Discussion by super-cooper** [keycloak/keycloak#15073] [Q&A] [answered]
I'm in the process of upgrading Keycloak from v16 to v17, and I'm having much trouble. I followed the upgrade guide and have been poring over the docs for several days now. We managed to get Keycloak up and running with an image that extends the official Keycloak image, and we updated our custom plugin as well. Both of those live in [this repository](https://github.com/ChameleonCloud/keycloak-chameleon) if you want to take a look.

I believe the custom plugin is loaded, since I see it in the server info page and our custom theme loads, but the only realm which loads is Master. Along with that, none of our clients or users show up, even though I can see them in our database. I compared the realm configuration on our production deployment (still running v16) and they look comparable. I also ensured that `ENABLED = true` for the missing realm, and I double checked the debug logs to ensure that Keycloak was pointed at the right database.

Any idea why this might be happening? Nothing about this was in the upgrading guides.

**Comment by maxcoolmakarov** [comment]
Met the same problem now. Have you solved it?

**Reply by super-cooper** [reply, marked as answer]
I believe it was something silly that was my fault. I don't remember at this point, but it was something like us pointing the server at the wrong database

**Reply by maxcoolmakarov** [reply]
I also found out that it was a mistake in DB hostname and keycloak used a temporary created db.
Thank you for your reply
